### PR TITLE
Setting cookies on hybrid remote webview instead of loading page through front door

### PR DESCRIPTION
**Pull request by wmathurin**
### New class: `SalesforceWebviewCookieManager`
- Similar to the same class in SalesforceMobileSDK-Android or MP's `CookieStoreManager` than SApp's `BridgeCookieManager`
- Gets SIDs from the user's credentials and sets them in the `WKWebsiteDataStore.default().httpCookieStore`

### Modified `SFHybridViewController`
- it no longer uses front-door to load remote app
- it calls `SalesforceWebviewCookieManager`'s `setCookies` method when receiving after authentication/refresh

### Manual testing
Used two apps on Mobile SDK cs4.com test org
- `apex/AccountEditor` which is the AccountEditor uploaded as hybrid remote app: it uses Mobile SDK Cordova plugins to do network operations - I had to change the `AccountEditor` visual force page to use `cordova.js` from the `cordovaios` resource (it was using the `cordovaandroid` previously)
- `apex/AccountList` which is a true Visual Force app that shows a list of accounts and allow navigation to their detail screen

#### Testing steps with `apex/AccountEditor`
- login to hybrid remote app pointing to `apex/AccountEditor`
- do some operations (that talk to the server)
- from a browser in setup -> session management, kill the oauth2 parent session
- go back to application, do some operations
- check the log, it shows that a refresh took place but it was completely transparent to the user

#### Testing steps with `apex/AccountList`
- login to hybrid remote app pointing to `apex/AccountList`
- click on some account, it navigates to the account detail page
- from a browser in setup -> session management, kill the oauth2 parent session
- go back to application, navigate again to an account detail page
- check the log, it shows that a refresh followed by setCookies took place but they were completely transparent to the user

### CODE REVIEW

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m [hunk 1]
```diff
@@ -613,8 +641,7 @@ - (void) webView:(WKWebView *) webView decidePolicyForNavigationAction:(WKNaviga
             }
             shouldAllowRequest = NO;
         } else {
-            [self defaultWKNavigationHandling:webView decidePolicyForNavigationAction:navigationAction decisionHandler:decisionHandler];
```
Comment (bbirman) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m:616]: I don't know the background on `defaultWKNavigationHandling`, can the whole method go away now?

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SalesforceWebViewCookieManager.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        if let cookie = HTTPCookie(properties: properties) {
+            cookieStore.setCookie(cookie)
+            SFSDKHybridLogger.d(Self.self, message: "[\(Self.self) \(#function)]: setting \(cookieType) in domain:\(domain) with value:\(value)")
+        }
+    }
+    
+    private func getHttpsUrlFromDomain(domain: String) -> URL {
+        if domain.contains("://") {
+            return URL(string: domain) ?? URL(string: "https://\(domain)")!
```
Comment (bbirman) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SalesforceWebViewCookieManager.swift:104]: I was wondering how this was used and if the force-unwrap could be avoided, but I don't see where this is called from?

Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SalesforceWebViewCookieManager.swift:104]: It's not needed on iOS. It came over when I converted the code over from kotlin (see https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/ui/SalesforceWebViewCookieManager.kt#L90).

Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SalesforceWebViewCookieManager.swift:104]: Planning to clean up logging / add tests today.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m (hunk 1 above)
Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m:616]: Hmm not sure - I think my refactoring was too aggressive.

Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m]: Reverted that change with https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid/pull/256/commits/c02fde85e24c6db99f33e743fd2656f642bb8565

**Review by bbirman — APPROVED** [review]

**Comment by bbirman** [comment]
cool to see the whole loading flow get more streamlined!

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m [hunk 3]
```diff
@@ -394,50 +389,20 @@ - (NSString *)sfHybridViewUserAgentString
     return userAgentString;
 }
 
-- (NSURL *)frontDoorUrlWithReturnUrl:(NSString *)returnUrlString returnUrlIsEncoded:(BOOL)isEncoded createAbsUrl:(BOOL)createAbsUrl
+- (NSURL *)absoluteUrlWithUrl:(NSString *)url
```
Comment (brandonpage) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m:392]: This is correct, but wow Objective-C naming is so dumb sometimes lol

**Review by brandonpage — COMMENTED** [review]
👍

**Pull request merged**